### ISSUE TITLE
x/tools/gopls: documentation of supported LSP features

### ISSUE THREAD

**Issue by jmhodges** [golang/go#47306]
I could totally believe I just wasn't able to find this (and maybe this ticket would then be for me to go make the linkages better).

It would be nice to have documentation on what features of the LSP protocol are supported (e.g. what kinds of refactoring, etc.). The [rust-analyzer one](https://rust-analyzer.github.io/manual.html#features) is an okay version. (It's currently got a bug where its hierarchies got flattened all to one, but hopefully that'll be fixed soon)

It seems like the [Features markdown file](https://github.com/golang/tools/blob/6e9046bfcd34178dc116189817430a2ad1ee7b43/gopls/doc/features.md) would be a good place for this

**Comment by thanm** [comment]
Would https://go.googlesource.com/tools/+/refs/heads/master/gopls/doc/commands.md be the sort of documentation you are looking for?

**Comment by jmhodges** [comment]
Yeah, I do believe so! I don't see it linked in [the README](https://github.com/golang/tools/blob/6e9046bfcd34178dc116189817430a2ad1ee7b43/gopls/README.md), which is how I missed it, I think

**Comment by jmhodges** [comment]
I'm not sure if it's better to put that file into Features or to link to both from the README. I _think_ it's better to combine them, but I'm not sure if there's some autogeneration or some such that makes that difficult or whatever.

**Comment by thanm** [comment]
@stamblerre per owners

**Comment by stamblerre** [comment]
The commands.md file specifies the supported commands through the [executeCommand request](https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#workspace_executeCommand), so it's not a comprehensive list. We can definitely autogenerate the set of supported features, but it's probably not something that we will prioritize for a while.
